### PR TITLE
Convert AuthToken criteria queries in UserController to use data API

### PR DESCRIPTION
**Pull request by sjrd218**
Convert AuthToken criteria queries in UserController to use data API

### CODE REVIEW

**Review by imad6639 — APPROVED** [review]

File: rundeckapp/src/main/groovy/org/rundeck/app/data/RdPageable.groovy [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.rundeck.app.data
+
+import org.rundeck.app.data.model.v1.page.Pageable
+
+class RdPageable implements Pageable{
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/app/data/RdPageable.groovy:5]: by default we should use `@CompileStatic` on all new groovy classes

File: rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy [hunk 2]
```diff
@@ -107,6 +109,16 @@ class GormTokenDataProvider implements TokenDataProvider {
         tokens
     }
 
+    @Override
+    List<AuthenticationToken> findAllByUser(String userId) {
+        List<AuthenticationToken> tokens = []
+        List<AuthToken> authTokens = AuthToken.findAllByUser(User.get(userId.toLong()))
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy:115]: should validate User.get return value

File: rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy [hunk 3]
```diff
@@ -161,17 +173,137 @@ class GormTokenDataProvider implements TokenDataProvider {
...
+        }
+    }
+
+    @Override
+    @GrailsCompileStatic(TypeCheckingMode.SKIP)
+    Integer countTokensByType(AuthenticationToken.AuthTokenType type) {
+        return AuthToken.createCriteria().count {
+            or {
+                eq("type", type)
+                isNull("type")   //type could be null if token was created before we recorded token types
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy:252]: I think we should remove support for "legacy" user tokens at some point.

File: rundeckapp/src/main/groovy/org/rundeck/app/data/RdPageable.groovy (hunk 1 above)
Comment (sjrd218) [reply rundeckapp/src/main/groovy/org/rundeck/app/data/RdPageable.groovy:7]: done

File: rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy (hunk 3 above)
Comment (sjrd218) [reply rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy:254]: Agreed, unless someone is upgrading from a very legacy version we shouldn't need to check this. Not sure what the priority would be on removing the extra check.

Comment (gschueler) [reply rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy:254]: i can backlog it

File: rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy [hunk 4]
```diff
@@ -161,17 +175,137 @@ class GormTokenDataProvider implements TokenDataProvider {
...
     }
 
     @Override
-    AuthenticationToken tokenLookupWithType(final String token, AuthenticationToken.AuthTokenType type){
-        def authToken = AuthToken.tokenLookup(token, type);
-        return authToken ?: null
+    @GrailsCompileStatic(TypeCheckingMode.SKIP)
+    Integer countTokensByUser(String userId) {
+        return AuthToken.createCriteria().count {
+            eq("user", User.get(userId.toLong()))
```
Comment (gschueler) [inline rundeckapp/src/main/groovy/org/rundeck/app/data/providers/GormTokenDataProvider.groovy:228]:
```suggestion
        User user = User.get(userId.toLong())
        if(!user) throw new DataAccessException("Couldn't find user: ${userId}")
        return AuthToken.createCriteria().count {
            eq("user", user)
```

**Review by gschueler — APPROVED** [review]

**Pull request merged**